geode-core/src/distributedTest/java/org/apache/geode/cache/management/MemoryThresholdsDUnitTest.java

```diff
@@ -2509,27 +2509,33 @@ public void testCriticalMemoryEventTolerance() {
     vm.invoke(new SerializableCallable() {
       @Override
       public Object call() throws Exception {
-        int defaultTolerance = 1;
         HeapMemoryMonitor.setTestDisableMemoryUpdates(false);
         GemFireCacheImpl cache = (GemFireCacheImpl) getCache();
         InternalResourceManager irm = cache.getInternalResourceManager();
         HeapMemoryMonitor hmm = irm.getHeapMonitor();
         hmm.setTestMaxMemoryBytes(100);
         HeapMemoryMonitor.setTestBytesUsedForThresholdSet(1);
         irm.setCriticalHeapPercentage(95);
-        for (int i = 0; i < defaultTolerance; i++) {
+        int previousMemoryStateChangeTolerance = hmm.getMemoryStateChangeTolerance();
+        try {
+          int memoryStateChangeTolerance = 3;
```

Comment:
It looks like there is a bit of duplication in this block and around line 2555. Is there an opportunity to refactor?

Reply:
I'll look into this...these tests were here prior to this PR - I just changed the value of the tolerance, but this is as good as any time to refactor these to be cleaner.  Because I wrapped the code in a `try` `finally` block, this shows up in GitHub as a big delete and add, which is somewhat misleading...